book/extend-admin.rst

```diff
@@ -0,0 +1,816 @@
+  uses this value as the header for the given column in the list.
+- The ``searchability`` attribute describes if the value of this property is used by the search field in the list.
+- Finally the ``type`` attribute allows to define how to display the content of this property. In the above example it
+  is used to display the datetime value in the localization of the user. There is an extension point for these types,
+  which allows to add more of them via JS.
+
+In addition to these attributes the ``property`` tag has some sub tags as well. This includes the ``field-name``
+telling the ``ListBuilder`` how the column holding the value in the database is called, and the ``entity-name``
+describing which entity holds the property. Based on this information the ``ListBuilder`` can build a very efficient
+query.
+
+The ``Controller`` returning the data from the ``ListBuilder`` makes us of the `FOSRestBundle`_ as well. The
+``cgetAction`` makes use of the ``FieldDescriptorFactory`` to load the information written in the above XML file. It
+then uses the ``DoctrineListBuilderFactory`` to get an instance of a ``DoctrineListBuilder``, which implements the
+logic to load data in an efficient way from the database. The ``RestHelper`` helps to set certain parameters of the
+``ListBuilder`` from the HTTP request, so that this code has not been copied over multiple times. Finally the
+``PaginatedRepresentation`` takes
+care of building an object representing the loaded data and enhance it with information like how many result exist in
+total. This object will be serialized by the ``handleView`` method of the `FOSRestBundle`_. The following code shows
+a controller doing what has just been described.
+
+.. code-block:: php
+
+    <?php
+
+    namespace App\Controller;
+
+    use App\Entity\Event;
+    use FOS\RestBundle\Routing\ClassResourceInterface;
+    use FOS\RestBundle\View\View;
+    use FOS\RestBundle\View\ViewHandlerInterface;
+    use Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactoryInterface;
+    use Sulu\Component\Rest\ListBuilder\Metadata\FieldDescriptorFactoryInterface;
+    use Sulu\Component\Rest\ListBuilder\PaginatedRepresentation;
+    use Sulu\Component\Rest\RestHelperInterface;
+    use Symfony\Component\HttpFoundation\Response;
+
+    class EventController implements ClassResourceInterface
+    {
+        /**
+         * @var ViewHandlerInterface
+         */
+        private $viewHandler;
+
+        /**
+         * @var FieldDescriptorFactoryInterface
+         */
+        private $fieldDescriptorFactory;
+
+        /**
+         * @var DoctrineListBuilderFactoryInterface
+         */
+        private $listBuilderFactory;
+
+        /**
+         * @var RestHelperInterface
+         */
+        private $restHelper;
+
+        public function __construct(
+            ViewHandlerInterface $viewHandler,
+            FieldDescriptorFactoryInterface $fieldDescriptorFactory,
```

Comment:
This line causes the website not to work, because this controller will be registered, but the `FieldDescriptorFactoryInterface` does not exist. There are multiple solutions to this problem:

1. We add [this configuration from the workshop repo](https://github.com/sulu/sulu-workshop-symfony-live-berlin-2019/blob/607ac5b/config/services.yaml#L48) to the skeleton. Then we decide that all controllers for the Admin have to be located there, and that they cannot be used in a website context.
2. Move the `FieldDescriptorInterface` somewhere else. I could imagine to have a `SuluListBuilderBundle` or something like that containing all the `ListBuilder` stuff being registered in both the `admin` and `website` context.
3. Just make the default value of this line `null`, then Symfony won't complain, however, if the controller is still called there might be problems because this dependency is missing (which we either can hope that it just does not happen, or it has to be handled in this controller somehow).

Reply:
@alexander-schranz @wachterjohannes @nnatter ???

Reply:
1 + 2 💪